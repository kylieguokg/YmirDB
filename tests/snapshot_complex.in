set h 1
set g 1
set f 1
set e 1
set d 1
list entries
snapshot
set c f g 2
set b d e 2
list entries
snapshot
set a 3 c b h 2
list entries
snapshot
del a
del b
del c
del d
del e
del f
del g
del h
list entries
checkout 2
list entries
purge h
purge g
purge f
purge e
purge d
purge c
purge b
purge a
list entries
checkout 3
list entries
set a 3 c b h 2
list entries
sum h
sum g
sum f
sum e
sum d
sum c
sum b
sum a
pop c
list entries
forward a
forward b
forward c
forward d
forward e
forward f
forward g
forward h
backward a
backward b
backward c
backward d
backward e
backward f
backward g
backward h
sum h
sum g
sum f
sum e
sum d
sum c
sum b
sum a
del a
list entries
drop 3
list snapshots
purge b
purge c
checkout 2
list entries
checkout 1
list entries
rollback 2
list snapshots
list entries
bye
