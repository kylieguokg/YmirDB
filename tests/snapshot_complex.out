> ok

> ok

> ok

> ok

> ok

> d [1]
e [1]
f [1]
g [1]
h [1]

> saved as snapshot 1

> ok

> ok

> b [d e 2]
c [f g 2]
d [1]
e [1]
f [1]
g [1]
h [1]

> saved as snapshot 2

> ok

> a [3 c b h 2]
b [d e 2]
c [f g 2]
d [1]
e [1]
f [1]
g [1]
h [1]

> saved as snapshot 3

> ok

> ok

> ok

> ok

> ok

> ok

> ok

> ok

> no entries

> ok

> b [d e 2]
c [f g 2]
d [1]
e [1]
f [1]
g [1]
h [1]

> not permitted

> not permitted

> not permitted

> not permitted

> not permitted

> not permitted

> not permitted

> ok

> b [d e 2]
c [f g 2]
d [1]
e [1]
f [1]
g [1]
h [1]

> ok

> b [d e 2]
c [f g 2]
d [1]
e [1]
f [1]
g [1]
h [1]

> ok

> a [3 c b h 2]
b [d e 2]
c [f g 2]
d [1]
e [1]
f [1]
g [1]
h [1]

> 1

> 1

> 1

> 1

> 1

> 4

> 4

> 14

> f

> a [3 c b h 2]
b [d e 2]
c [g 2]
d [1]
e [1]
f [1]
g [1]
h [1]

> b, c, d, e, g, h

> d, e

> g

> nil

> nil

> nil

> nil

> nil

> nil

> a

> a

> a, b

> a, b

> nil

> a, c

> a

> 1

> 1

> 1

> 1

> 1

> 3

> 4

> 13

> ok

> b [d e 2]
c [g 2]
d [1]
e [1]
f [1]
g [1]
h [1]

> ok

> 2
1

> ok

> ok

> ok

> d [1]
e [1]
f [1]
g [1]
h [1]

> ok

> d [1]
e [1]
f [1]
g [1]
h [1]

> ok

> 2
1

> d [1]
e [1]
f [1]
g [1]
h [1]

> bye
